### ISSUE TITLE
Signup is not working properly.

### ISSUE THREAD

**Issue by LilPhyziX** [daveh/php-signup-login#14]
Hello. When i try to register, i am greeted with this message: Field 'id' doesn't have a default value 1364

I dont know what this means but all i know is its preventing the details to go into the database. 

Any help would be greatly appreciated. 

Thanks in advance!

**Comment by daveh** [comment]
It looks like you don't have AUTO_INCREMENT set on your id column in the database - this assigns IDs automatically when you insert a new record.

**Comment by LilPhyziX** [comment]
I do have auto increment on though. However i dont see anything in the code that inserts anything into the id

**Comment by daveh** [comment]
Then you don't need to include the id in the list of fields in the insert statement, or in the list of values. New records will be assigned a new value automatically.

**Comment by LilPhyziX** [comment]
Alright. I fixed the signup, however the login is not working now.

**Comment by LilPhyziX** [comment]
Here is the code for the signin page:

<?php

$is_invalid = false;

if ($_SERVER["REQUEST_METHOD"] === "POST") {
    
    $mysqli = require __DIR__ . "/database.php";
    
    $sql = sprintf("SELECT * FROM cbs_user
                    WHERE email = '%s'",
                   $mysqli->real_escape_string($_POST["email"]));
    
    $result = $mysqli->query($sql);
    
    $user = $result->fetch_assoc();
    
    if ($user) {
        
        if (password_verify($_POST["password"], $user["password_hash"])) {
            
            session_start();
            
            session_regenerate_id();
            
            $_SESSION["user_id"] = $user["id"];
            
            header("Location: index.php");
            exit;
        }
    }
    
    $is_invalid = true;
}

?>

<!doctype html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>Sign In</title>
  <link rel="icon" type="image/x-icon" href="assets/images/favicon.ico">
  <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css" rel="stylesheet"
    integrity="sha384-rbsA2VBKQhggwzxH7pPCaAqO46MgnOM80zW1RWuH61DGLwZJEdK2Kadq2F9CUG65" crossorigin="anonymous">
  <script src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.11.6/dist/umd/popper.min.js"
    integrity="sha384-oBqDVmMz9ATKxIep9tiCxS/Z9fNfEXiDAYTujMAeBAsjFuCZSmKbSSUnQlmh/jp3"
    crossorigin="anonymous"></script>
  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.min.js"
    integrity="sha384-cuYeSxntonz0PPNlHhBs68uyIAVpIIOZZ5JqeqvYYIcEL727kskC66kF92t6Xl2V"
    crossorigin="anonymous"></script>
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css"
    integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU" crossorigin="anonymous">
  <link href="assets/css/style.css" rel="stylesheet">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/OwlCarousel2/2.3.4/assets/owl.carousel.min.css" />
  <script src="https://cdnjs.cloudflare.com/ajax/libs/OwlCarousel2/2.3.4/owl.carousel.min.js"></script>

</head>

<body>
  <header>
    <div class="container-fluid">
      <div class="row">
        <div class="col-md-12">
          <nav class="navbar navbar-expand-lg navbar-dark">
            <div class="container-fluid">
              <a class="navbar-brand" href="index.php">
                <img src="assets/images/logo.png" class="img-fluid" alt="img" />
                CherryBerry <span class="text-red"> Studios </span>
              </a>
              <button class="navbar-toggler" type="button" data-bs-toggle="collapse"
                data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false"
                aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
              </button>
              <div class="collapse navbar-collapse" id="navbarSupportedContent">
                <ul class="navbar-nav ms-auto mb-2 mb-lg-0">

                  <li class="nav-item">
                    <a class="nav-link" href="index.php">Home</a>
                  </li>
                  <li class="nav-item">
                    <a class="nav-link" href="about.html">About Us</a>
                  </li>


                  <li class="nav-item dropdown">
                    <a class="nav-link dropdown-toggle" href="#" role="button" aria-expanded="false">
                      Story
                    </a>
                    <ul class="dropdown-menu">
                      <li><a class="dropdown-item" href="mission.html">Mission</a></li>
                      <li>
                        <hr class="dropdown-divider">
                      </li>
                      <li><a class="dropdown-item" href="vision.html">Vision</a></li>
                      <li>
                        <hr class="dropdown-divider">
                      </li>
                      <li><a class="dropdown-item" href="objective.html">Objective</a></li>
                    </ul>
                  </li>


                  <li class="nav-item">
                    <a class="nav-link" href="community.html">Community</a>
                  </li>
                  <li class="nav-item">
                    <a class="nav-link" href="support.html">Support</a>
                  </li>

                </ul>
                <form class="d-flex ms-lg-5">

                  <a href="signup.html" class="btn btn-menu">Sign Up</a>
                </form>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </header>




  <div class="cover" style="background:url(assets/images/banner.jpg)">
    <div class="background-div pt-5 pb-5">
      <div class="container pt-5 pb-5">
        <div class="row pt-md-2 pb-md-3">
          <div class="col-md-12">
            <h4 class="head3 text-center"> Sign In To Your Account </h4>
            <p class="text-center pb-3">Let's get started with your account</p>
          </div>
          <div class="col-lg-6 offset-lg-3 col-md-8 offset-md-2 relative">

            <div class="card reg-form pt-3 pb-3">
              <div class="card-body">

<?php if ($is_invalid): ?>
        <em>Invalid login</em>
    <?php endif; ?>
    
                <form method="post">
                  <p class="mb-2 pt-1 label-fr">Username</p>
                  <div class="input-group mb-3 input-group-custom">
                    <span class="input-group-text input-group-custom-field">
                      <i class="far fa-user"></i>
                    </span>
                    <input type="text" class="form-control input-group-custom-field" placeholder="Username" name="email" id="email" value="<?= htmlspecialchars($_POST["email"] ?? "") ?>">
                  </div>
                  <p class="mb-2 pt-1 label-fr">Password</p>
                  <div class="input-group mb-3 input-group-custom">
                    <span class="input-group-text input-group-custom-field">
                      <i class="fa fa-lock"></i>
                    </span>
                    <input type="password" class="form-control input-group-custom-field" placeholder="XXXXXXX" name="password" id="password">
                  </div>
                  <div class="form-group pt-1">
                    <label>
                      <input type="checkbox" required>
                      <span class="rem"> Remember Me </span>
                    </label>
                  </div>
                  <div class="form-group text-center mb-4 mt-3">
                    <button type="button" class="btn btn1 ps-4 pe-4 pt-2 pb-2"> SIGN IN </button>
                  </div>
                  <div class="d-lg-flex justify-content-between links text-center f15">
                    <p class="mb-md-0 mb-1">
                      Don't have an account? <a href="signup.html" class="gray text-links">Sign Up</a>
                    </p>
                    <p class="mb-md-0 mb-1">
                      Forgot Password? <a href="forgot-password.html" class="gray text-links">Reset it</a>
                    </p>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>

  <footer>
    <div class="card nl-form">
    <h1>Subscribe to Our Newsletter</h1>
    <p>Stay updated with the latest news and promotions.</p>
    <form class="card-body" action="subscribe.php" method="post">
      <input class="form-control input-group-custom-field" type="email" name="email" placeholder="Enter your email" required>
      <button class="btn btn1" type="submit">Subscribe</button>
    </form>
    </div>
    <div class="container">
      <div class="row">
        <div class="col-md-6 col-lg-4 pt-4">
          <h5 class="pb-2">
            CherryBerry <span class="text-red">Studios</span>
          </h5>

          <p class="pe-xl-5">
            Duis posuere accumsan neque eu tristique. Sed at enim ultrices, cursus ipsum id, mattis massa.
          </p>
        </div>


        <div class="col-md-6 col-lg-2 pt-4">
          <h5 class="pb-2">
            Quick Links
          </h5>

          <ul class="list-inline list1">
            <li>
              <a href="index.html">
                Home
              </a>
            </li>
            <li>
              <a href="about.html">
                About Us
              </a>
            </li>
            <li>
              <a href="">
                Sign Up
              </a>
            </li>
          </ul>
        </div>
        <div class="col-md-6 col-lg-2 pt-4">
          <h5 class="pb-2">
            Quick Links
          </h5>

          <ul class="list-inline list1">
            <li>
              <a href="index.html">
                Home
              </a>
            </li>
            <li>
              <a href="about.html">
                About Us
              </a>
            </li>
            <li>
              <a href="">
                Sign Up
              </a>
            </li>
          </ul>
        </div>
        <div class="col-md-6 col-lg-4 pt-4">
          <h5 class="pb-2">
            Contact Us
          </h5>

          <ul class="list-inline list1">
            <li>
              <a href="mailto:support@cherryberrystudios.com">
                <i class="far fa-envelope"></i>
                support@cherryberrystudios.com 
              </a>
            </li>

          </ul>


          <ul class="list-inline  social-list2">
            <li class="list-inline-item">
              <a href="">
                <i class="fab fa-facebook-f"></i>
              </a>
            </li>
            <li class="list-inline-item">
              <a href="">
                <i class="fab fa-twitter"></i>
              </a>
            </li>
            <li class="list-inline-item">
              <a href="">
                <i class="fab fa-youtube"></i>
              </a>
            </li>
            <li class="list-inline-item">
              <a href="">
                <i class="fab fa-instagram"></i>
              </a>
            </li>
          </ul>

        </div>
      </div>
    </div>
  </footer>
  <div class="copyrights text-center p-3">
    <p class="m-0">
      ©2023 - CherryBerry Studios - All rights reserved.
    </p>
  </div>

    <div id="disclaimer" class="dis-popup">
        <p class="dis-text">This website uses cookies and collects diagnostic data to improve your browsing experience. By continuing to use this site, you consent to the use of cookies and data collection. For more details, please refer to our <a class="gray text-links" href="privacy.html">Privacy Policy</a>.</p>
        <button class="btn btn1 dis-btn" id="acceptButton">Accept & Continue</button>
    </div>

  <div class="loading-screen">
    <div class="loading-spinner-container">
        <img src="assets/images/ls.png" alt="Logo" class="ld-img">
        <div class="loading-spinner"></div>
    </div>
    <h1>Please Wait...</h1>
  </div>

  <script src="assets/js/core.js"></script>




</body>

</html>

**Comment by LilPhyziX** [comment]
Uhm. I dont think it was supposed to display the actual elements

**Comment by LilPhyziX** [comment]
I will just upload the file. #
[signin.php.txt](https://github.com/daveh/php-signup-login/files/12305287/signin.php.txt)

**Comment by daveh** [comment]
Please can you describe exactly what the problem is? Do you get any error messages?
